src/components/Layer/EditControl/EditControl.tsx

```diff
@@ -23,8 +25,13 @@ export function EditControl({
     }
   }
   return (
-    <div className={style.edit} onClick={editLayer}>
-      <Edit16 />
-    </div>
+    <Tooltip placement="top">
+      <TooltipTrigger asChild>
+        <div className={style.edit} onClick={editLayer}>
```

Comment:
The whole part wrapping icon is duplicated multiple times across different components.
I think it's a good time to extract everything but the icon and logic into a reusable component. Something like this:

```
type LayerActionButtonProps = {
  onClick: () => void;
  children: ReactElement;
  hint?: string;
  className?: string;
};

function LayerActionButton({
  onClick,
  hint,
  children,
  className,
}: LayerActionButtonProps) {
  return (
    <Tooltip placement="top">
      <TooltipTrigger asChild>
        <div className={clsx(style.actionButton, className)} onClick={onClick}>
          {children}
        </div>
      </TooltipTrigger>
      {hint ? <TooltipContent>{hint}</TooltipContent> : null}
    </Tooltip>
  );
}
```